SNOW-1643304 Record the `set_index` length checking behavior change in the changelog (#2180)
snowflakedb/snowpark-python: CHANGELOG.md

- Fixed a bug in `Series.reset_index(drop=True)` where the result name may be wrong.
- Fixed a bug in `Groupby.first/last` ordering by the correct columns in the underlying window expression.

#### Behavior Change

> - When performing `set_index` on a `Series` or `DataFrame` object whose length does not match with the new index's length, a `ValueError` is no longer raised. When the `Series`/`DataFrame` object is longer than the new index, the `Series`/`DataFrame`'s new index is filled with `NaN` values for the "extra" elements. When the `Series`/`DataFrame` object is shorter than the new index, the extra values in the new index are ignored—`Series` and `DataFrame` stay the same length `n`, and use only the first `n` values of the new index.

Reviewer: There's no `set_index` method for `Series`, so we should clarify with something like "setting an index by calling DataFrame.set_index, setting Series.index, or setting DataFrame.index"

Author: done!

> - When performing `set_index` on a `Series` or `DataFrame` object whose length does not match with the new index's length, a `ValueError` is no longer raised. When the `Series`/`DataFrame` object is longer than the new index, the `Series`/`DataFrame`'s new index is filled with `NaN` values for the "extra" elements. When the `Series`/`DataFrame` object is shorter than the new index, the extra values in the new index are ignored—`Series` and `DataFrame` stay the same length `n`, and use only the first `n` values of the new index.

Author: done! This is in `base.py`. I updated both `_set_index` and `_get_index` since I think `_get_index`'s docstring is the one being displayed for `Series.index` and `DataFrame.index`

- Fixed a bug in `Series.reset_index(drop=True)` where the result name may be wrong.
- Fixed a bug in `Groupby.first/last` ordering by the correct columns in the underlying window expression.

#### Behavior Change

> - When calling `DataFrame.set_index`, or setting `DataFrame.index` or `Series.index`, with a new index that does match the current length of the `Series`/`DataFrame` object does not match with the new index's length, a `ValueError` is no longer raised. When the `Series`/`DataFrame` object is longer than the new index, the `Series`/`DataFrame`'s new index is filled with `NaN` values for the "extra" elements. When the `Series`/`DataFrame` object is shorter than the new index, the extra values in the new index are ignored—`Series` and `DataFrame` stay the same length `n`, and use only the first `n` values of the new index.

Author: done!

#### Behavior Change

> - When performing `DataFrame.set_index`, or setting `DataFrame.index` or `Series.index` where the length of the `Series`/`DataFrame` object does not match with the new index's length, a `ValueError` is no longer raised. When the `Series`/`DataFrame` object is longer than the new index, the `Series`/`DataFrame`'s new index is filled with `NaN` values for the "extra" elements. When the `Series`/`DataFrame` object is shorter than the new index, the extra values in the new index are ignored—`Series` and `DataFrame` stay the same length `n`, and use only the first `n` values of the new index.

Reviewer: ```suggestion
- When calling `DataFrame.set_index`, or setting `DataFrame.index` or `Series.index`, with a new index that does match the current length of the `Series`/`DataFrame`, a `ValueError` is no longer raised. When the `Series`/`DataFrame` object is longer than the new index, the `Series`/`DataFrame`'s new index is filled with `NaN` values for the "extra" elements. When the `Series`/`DataFrame` object is shorter than the new index, the extra values in the new index are ignored—`Series` and `DataFrame` stay the same length `n`, and use only the first `n` values of the new index.
```

Author: bundled in with the new commit
